server/src/main/java/org/candlepin/policy/SystemPurposeComplianceRules.java

```diff
@@ -0,0 +1,214 @@
+                    List<String> addOns =
+                        Arrays.asList(product.getAttributeValue("addons").split("\\s*," + "\\s*"));
+                    for (String addOn: unsatisfiedAddons) {
+                        if (addOns.stream().anyMatch(str -> str.trim().equals(addOn))) {
+                            status.addCompliantAddOn(addOn, entitlement);
+                            addonsFound.add(addOn);
+                        }
+                    }
+                }
+                unsatisfiedAddons.removeAll(addonsFound);
+
+                if (product.hasAttribute("support_level")) {
+                    String sla = product.getAttributeValue("support_level");
+                    if (StringUtils.isNotEmpty(preferredSla) && sla.equalsIgnoreCase(preferredSla)) {
+                        status.addCompliantSLA(sla, entitlement);
+                    }
+                    else {
+                        status.addNonPreferredSLA(preferredSla, sla, entitlement);
+                    }
+                }
+
+                if (product.hasAttribute("usage")) {
+                    String usage = product.getAttributeValue("usage");
+                    if (StringUtils.isNotEmpty(preferredUsage) && usage.equalsIgnoreCase(preferredUsage)) {
+                        status.addCompliantUsage(usage, entitlement);
+                    }
+                    else {
+                        status.addNonPreferredUsage(preferredUsage, usage, entitlement);
+                    }
+                }
+            }
+        }
+
+        // check if there are unsatisfied role, add ons, slas and usage
+        if (StringUtils.isNotEmpty(consumer.getRole()) &&
+            !status.getCompliantRole().containsKey(consumer.getRole())) {
+            status.setNonCompliantRole(consumer.getRole());
+        }
+        if (CollectionUtils.isNotEmpty(consumer.getAddOns())) {
+            for (String addOn : consumer.getAddOns()) {
+                if (!status.getCompliantAddOns().containsKey(addOn)) {
+                    status.addNonCompliantAddOn(addOn);
+                }
+            }
+        }
+        if (StringUtils.isNotEmpty(consumer.getServiceLevel()) && status.getCompliantSLA().isEmpty()) {
+            status.setNonCompliantSLA(consumer.getServiceLevel());
+        }
+        if (StringUtils.isNotEmpty(consumer.getUsage()) && status.getCompliantUsage().isEmpty()) {
+            status.setNonCompliantUsage(consumer.getUsage());
+        }
+
+        if (currentCompliance) {
+            applyStatus(consumer, status, updateConsumer);
+        }
+
+        return status;
+    }
+
+    public void applyStatus(Consumer c, SystemPurposeComplianceStatus status, boolean updateConsumer) {
+        String newHash = getComplianceStatusHash(status, c);
+        boolean complianceChanged = !newHash.equals(c.getComplianceStatusHash());
+        if (complianceChanged) {
+            log.debug("Compliance has changed, sending Compliance event.");
```

Comment:
Should change this string to say Syspurpose Compliance since otherwise would make for a confusing debug line.